src/TwigComponent/src/Event/PostMountEvent.php

```diff
@@ -21,6 +22,7 @@ final class PostMountEvent extends Event
     public function __construct(
         private object $component,
         private array $data,
+        private ComponentMetadata $metadata,
         private array $extraMetadata = [],
     ) {
```

Comment:
This is a BC break, though almost definitely minor. But we should handle it. General steps:

A) Remove the types from the 3rd and 4th argument
B) If the 3rd argument is an array, trigger a deprecation - something like:

> In TwigComponent 3.0, the 3rd argument of `PostMountEvent` will be a `ComponentMetadata` object.

And, in this case, set the 3rd arg `$metadata` will be set onto the `$extraMetadata` property (so, also, we will need to remove the `private` from args 3 and 4 and set them as normal properties).

Also in this situation, set the `$metadata` property to null. We'll need to make the return type on `getMetadata()` nullable for now.

Reply:
Yup. So, use an if statement and trigger a deprecation if they pass the "old" way of doing things. But then, if they just pass some bad value - e.g. they pass a string to the 3rd arg - which was NEVER legal - check that and throw an exception.

Reply:
That also would require to check in LiveComponent if the installed TwigComponent has the good version (as they are two different packages)